### DISCUSSION TITLE
Field mask

### DISCUSSION THREAD

**Discussion by Hedgehogues** [fastapi/fastapi#7908] [Questions]
### First check

* [x] I used the GitHub search to find a similar issue and didn't find it.
* [x] I searched the FastAPI documentation, with the integrated search.
* [x] I already searched in Google "How to X in FastAPI" and didn't find any information.

### Description

Does has fast api a field masks like a flask-rest-plus https://flask-restplus.readthedocs.io/en/stable/mask.html ?

**Comment by dmontagu** [comment]
I'm not aware of anything like this. This could definitely be implemented, you'd have to do it manually though.

------

It appears to me that it doesn't lend itself to a very type-safe way of interacting with the data submitted to your endpoint, which is probably why it is difficult to follow this pattern with FastAPI out of the box.

**Comment by Hedgehogues** [comment]
@dmontagu Some guys say me about [this](https://fastapi.tiangolo.com/tutorial/graphql/). This component has the same functional, I think. I don't touch this one yet. What are you think about it?

**Comment by dmontagu** [comment]
I haven’t used GraphQL too much but from my understanding that’s the modern way to serve an API returning partial data along the lines of the project you linked.

I have no idea what it looks like to write a GraphQL-based backend though; if anyone knows (or has written) a good blog article about creating a graphql service in FastAPI or starlette I’d love to read it!

**Comment by Hedgehogues** [comment]
Field mask is graphQL for poored.

Probably, this case it solves in starlette or pydantic.

**Comment by tiangolo** [comment]
Thanks for the help here @dmontagu ! :cake: :bowing_man: 

And thanks @Hedgehogues for reporting back and closing the issue :+1:
